### DISCUSSION TITLE
buildsys build logs

### DISCUSSION THREAD

**Discussion by trebonian** [bottlerocket-os/twoliter#406] [Q&A]
Buildsys spawns a "docker buildx build" for each package. When docker build runs interactively, it writes a log trace to stdout and stderr. I can't find how to redirect the buildx output to a log file. It would be nice to do it for a given package. Is there a parameter or an Env argument I can give to redirect the buildx output to a file?

**Comment by bcressey** [comment]
There's no way to do that currently, but there's a close equivalent for successful builds: the output log maintained by `cargo`, which you can find under the target directory at the root of the checkout.

For example, for the kernel-6.1 package, the logs can be found here:
```
target/*/debug/build/kernel-6_1-*/output
```

Because it's done by `cargo` it uses the crate name of the package, which is usually the same as what's in the directory tree. The exceptions are packages with dots in the name, where the dot is replaced with an underscore. Hence `kernel-6.1` becomes `kernel-6_1` in the case above.

**Comment by trebonian** [comment]
Is there a way to force the "—verbose" flag to the cargo build being used by twoliter? Even better if I could do this for the "build-package" task in the cargo make used by twoliter.
I would like to tail the build log for particular packages while its building...
